sbt-plugin-test/build.sbt

```diff
@@ -69,33 +67,46 @@ lazy val noDOM = project.settings(baseSettings: _*).
       }
   )))
 
-lazy val testFramework = crossProject.crossType(CrossType.Pure).
-  settings(versionSettings: _*).
-  settings(name := "Dummy cross JS/JVM test framework").
-  jsSettings(
+lazy val testFrameworkCommonSettings = Def.settings(
+  versionSettings,
+  name := "Dummy cross JS/JVM test framework",
+  scalaSource in Compile := baseDirectory.value.getParentFile / "src/main/scala"
+)
+
+lazy val testFrameworkJS = project.in(file("testFramework/.js")).
+  enablePlugins(ScalaJSPlugin).
+  settings(
+    testFrameworkCommonSettings,
     libraryDependencies +=
       "org.scala-js" %% "scalajs-test-interface" % scalaJSVersion
-  ).
-  jvmSettings(
+  )
+
+lazy val testFrameworkJVM = project.in(file("testFramework/.jvm")).
+  settings(
+    testFrameworkCommonSettings,
     libraryDependencies ++= Seq(
         "org.scala-sbt" % "test-interface" % "1.0",
         "org.scala-js" %% "scalajs-stubs" % scalaJSVersion % "provided"
     )
   )
 
-lazy val testFrameworkJS = testFramework.js
-lazy val testFrameworkJVM = testFramework.jvm
+lazy val multiTestCommonSettings = Def.settings(
+  unmanagedSourceDirectories in Compile +=
+    baseDirectory.value.getParentFile / "shared/src/main/scala",
+  unmanagedSourceDirectories in Test +=
+    baseDirectory.value.getParentFile / "shared/src/test/scala",
+
+  testFrameworks ++= Seq(
+      TestFramework("sbttest.framework.DummyFramework"),
+      TestFramework("inexistent.Foo", "another.strange.Bar")
+  )
+)
 
-lazy val multiTest = crossProject.
```

Comment:
Just to check my understanding: We cannot depend on sbt-crossproject for this, since it will depend on SJS?

Reply:
That's right. Or to be more precise, `sbt-scalajs-crossproject` will (already does) depend on SJS. `sbt-crossproject` per se only depends on `sbt-platform-deps`, but also does not provide `JSPlatform` (which is provided by `sbt-scalajs-crossproject`).